server/src/test/java/org/elasticsearch/snapshots/SnapshotShutdownProgressTrackerTests.java

```diff
@@ -106,7 +195,13 @@ void simulateShardSnapshotsCompleting(SnapshotShutdownProgressTracker tracker, i
             }
             logger.info("---> Generated shard snapshot status in stage (" + status.getStage() + ") for switch case (" + (i % 4) + ")");
             tracker.decNumberOfShardSnapshotsInProgress(dummyShardId, dummySnapshot, status);
+            // Wait for the next periodic log message to include the new completion stats.
```

Comment:
You're not waiting. You're advancing the time and then running all the queued tasks so that the next log message, which is already scheduled in the queue, will be executed.